### PR TITLE
sglx_process_probe .json & .env is ready | tested running on EC2 as well

### PR DESCRIPTION
**Pull request by drewyangdev**

### CODE REVIEW

File: ecephys_spike_sorting/scripts/create_input_json.py [hunk 1]
```diff
@@ -57,32 +66,39 @@ def createInputJson(output_file,
...
+    # kilosort_repository = r'C:\Users\labadmin\Documents\jic\KS3_fork\Kilosort2'
+    # KS2ver = '2.0'      # must equal '3.0', '2.5' or '2.0', and match the kiilosort_repository
+    kilosort_repository = kilosort_repository
+    KS2ver = KS2ver
+
     # KS 3.0 does not yet output pcs.
     if KS2ver == '3.0':
         include_pcs = False  # set to false for KS2ver = '3.0'
     else:
         include_pcs = True
```
Comment (dimitri-yatsenko) [inline ecephys_spike_sorting/scripts/create_input_json.py]:
```suggestion
    include_pcs = KS2ver != '3.0'
```

File: ecephys_spike_sorting/scripts/sglx_process_probe.py [hunk 2]
```diff
@@ -31,6 +42,19 @@ def run_probe(prb, json_directory, npx_directory,
               c_Waves_snr_um=160,
               ni_present=True,
               ni_extract_string=None):
+    # load external tool path from .env if not given from .json
+    if dot_env_path and os.path.exists(dot_env_path):
+        load_dotenv(dot_env_path)
+        ecephys_directory=ecephys_directory if ecephys_directory else os.getenv('ecephys_directory')
+        kilosort_repository=kilosort_repository if kilosort_repository else os.getenv('kilosort_repository')
+        KS2ver=KS2ver if KS2ver else os.getenv('KS2ver')
```
Comment (dimitri-yatsenko) [inline ecephys_spike_sorting/scripts/sglx_process_probe.py]:
```suggestion
        KS2ver = KS2ver or os.getenv('KS2ver')
```

File: ecephys_spike_sorting/scripts/sglx_process_probe.py [hunk 3]
```diff
@@ -31,6 +42,19 @@ def run_probe(prb, json_directory, npx_directory,
               c_Waves_snr_um=160,
               ni_present=True,
               ni_extract_string=None):
+    # load external tool path from .env if not given from .json
+    if dot_env_path and os.path.exists(dot_env_path):
+        load_dotenv(dot_env_path)
+        ecephys_directory=ecephys_directory if ecephys_directory else os.getenv('ecephys_directory')
+        kilosort_repository=kilosort_repository if kilosort_repository else os.getenv('kilosort_repository')
```
Comment (dimitri-yatsenko) [inline ecephys_spike_sorting/scripts/sglx_process_probe.py]:
```suggestion
        kilosort_repository = kilosort_repository or os.getenv('kilosort_repository')
```

File: ecephys_spike_sorting/scripts/create_input_json.py [hunk 4]
```diff
@@ -57,32 +66,39 @@ def createInputJson(output_file,
...
-    npy_matlab_repository = r'C:\Users\labadmin\Documents\jic\npy-matlab-master'
-    catGTPath = r'C:\Users\labadmin\Documents\jic\CatGT-win'
-    tPrime_path=r'C:\Users\labadmin\Documents\jic\TPrime-win'
-    cWaves_path=r'C:\Users\labadmin\Documents\jic\C_Waves-win'
-    
+    #npy_matlab_repository = r'C:\Users\labadmin\Documents\jic\npy-matlab-master'
+    #catGTPath = r'C:\Users\labadmin\Documents\jic\CatGT-win'
+    #tPrime_path=r'C:\Users\labadmin\Documents\jic\TPrime-win'
+    #cWaves_path=r'C:\Users\labadmin\Documents\jic\C_Waves-win'
+    npy_matlab_repository = npy_matlab_repository
```
Comment (dimitri-yatsenko) [inline ecephys_spike_sorting/scripts/create_input_json.py]: What do these assignments do?

**Review by dimitri-yatsenko — CHANGES_REQUESTED** [review]

File: ecephys_spike_sorting/scripts/create_input_json.py [hunk 5]
```diff
@@ -57,32 +66,39 @@ def createInputJson(output_file,
...
+
     # location of kilosor respository and kilosort version
 
-    kilosort_repository = r'C:\Users\labadmin\Documents\jic\KS3_fork\Kilosort2'
-    KS2ver = '2.0'      # must equal '3.0', '2.5' or '2.0', and match the kiilosort_repository
-    
+    # kilosort_repository = r'C:\Users\labadmin\Documents\jic\KS3_fork\Kilosort2'
+    # KS2ver = '2.0'      # must equal '3.0', '2.5' or '2.0', and match the kiilosort_repository
+    kilosort_repository = kilosort_repository
+    KS2ver = KS2ver
```
Comment (dimitri-yatsenko) [inline ecephys_spike_sorting/scripts/create_input_json.py]: What's the purpose of these assignments?

**Review by ttngu207 — COMMENTED** [review]
Do we want to commit the `.env` and the input `.json` file?

File: ecephys_spike_sorting/scripts/create_input_json.py (hunk 4 above)
Comment (drewyangdev) [reply ecephys_spike_sorting/scripts/create_input_json.py]: The original code's path assignments were hardcoded, but I need them to be passed through the arguments. Those are just temp reminders for myself while I was coding, I'll remove them and add comments instead.

**Comment by drewyangdev** [comment]
> Do we want to commit the `.env` and the input `.json` file?

Was thinking to share those files with you in this way, I can take them out.

**Review by ttngu207 — COMMENTED** [review]
Also, I think `dot_env_path` should not be an argument to be passed in the `run_probe()` part. At the step of calling `run_probe`, the users would not know (and probably should not have to know) about all of the package setup (in the .env file). 
I think those can just assumed to be already specified as environment variables, and we can document those in the README

**Comment by drewyangdev** [comment]
> Also, I think `dot_env_path` should not be an argument to be passed in the `run_probe()` part. At the step of calling `run_probe`, the users would not know (and probably should not have to know) about all of the package setup (in the .env file).
> I think those can just assumed to be already specified as environment variables, and we can document those in the README

Sure, I'll update it.

File: ecephys_spike_sorting/scripts/create_input_json.py (hunk 1 above)
Comment (drewyangdev) [reply ecephys_spike_sorting/scripts/create_input_json.py]: https://github.com/vathes/ecephys_spike_sorting/pull/2/commits/6dbe139c9eb62b8ed8132b88bbd40f446d535e63

File: ecephys_spike_sorting/scripts/sglx_process_probe.py (hunk 2 above)
Comment (drewyangdev) [reply ecephys_spike_sorting/scripts/sglx_process_probe.py]: https://github.com/vathes/ecephys_spike_sorting/pull/2/commits/6dbe139c9eb62b8ed8132b88bbd40f446d535e63

File: ecephys_spike_sorting/scripts/create_input_json.py (hunk 4 above)
Comment (drewyangdev) [reply ecephys_spike_sorting/scripts/create_input_json.py]: https://github.com/vathes/ecephys_spike_sorting/pull/2/commits/6dbe139c9eb62b8ed8132b88bbd40f446d535e63

File: ecephys_spike_sorting/scripts/create_input_json.py (hunk 5 above)
Comment (drewyangdev) [reply ecephys_spike_sorting/scripts/create_input_json.py]: https://github.com/vathes/ecephys_spike_sorting/pull/2/commits/6dbe139c9eb62b8ed8132b88bbd40f446d535e63

File: ecephys_spike_sorting/scripts/sglx_process_probe.py [hunk 6]
```diff
@@ -31,6 +41,19 @@ def run_probe(prb, json_directory, npx_directory,
...
+        load_dotenv(dot_env_path)
+        ecephys_directory=ecephys_directory or os.getenv('ecephys_directory')
+        kilosort_repository=kilosort_repository or os.getenv('kilosort_repository')
+        KS2ver=KS2ver or os.getenv('KS2ver')
+        npy_matlab_repository=npy_matlab_repository or os.getenv('npy_matlab_repository')
+        catGTPath=catGTPath or os.getenv('catGTPath')
+        tPrime_path=tPrime_path or os.getenv('tPrime_path')
+        cWaves_path=cWaves_path or os.getenv('cWaves_path')
+        kilosort_output_tmp=kilosort_output_tmp or os.getenv('kilosort_output_tmp') 
+
```
Comment (ttngu207) [inline ecephys_spike_sorting/scripts/sglx_process_probe.py:56]: I think this block should be in `createInputJson` and not in `run_probe`. The point is that `createInputJson` should have special logic handling loading the paths of the required packages (if not specified), and `run_probe` is an "end-user" routine assuming all of the package configuration has been taken care of

Comment (drewyangdev) [reply ecephys_spike_sorting/scripts/sglx_process_probe.py:56]: Good point, I totally agree with you, but this update will require changes in the mathwork project's matlab code as well. I'm still focusing to make the whole pipeline running successfully at least once, I think this is something more important for now. Can we please keep an issue of this point and come back later?

Comment (ttngu207) [reply ecephys_spike_sorting/scripts/sglx_process_probe.py:56]: yes, certainly, I'll approve this PR
(I don't have write-access to merge this)

**Review by ttngu207 — APPROVED** [review]

File: ecephys_spike_sorting/scripts/sglx_process_probe.py (hunk 6 above)
Comment (drewyangdev) [reply ecephys_spike_sorting/scripts/sglx_process_probe.py:56]: Thanks, I'll merge it then

**Pull request merged**